proto-lens/src/Data/ProtoLens/Encoding/Bytes.hs

```diff
@@ -57,11 +58,27 @@ anyBits = loop 0 0
             b <- anyWord8
             loop (w .|. shiftL (fromIntegral b) n) (n+8)
 
-putVarInt :: Word64 -> Builder
-putVarInt n
-    | n < 128 = Builder.word8 (fromIntegral n)
-    | otherwise = Builder.word8 (fromIntegral $ n .&. 127 .|. 128)
-                      <> putVarInt (n `shiftR` 7)
+-- | A wrapper around a 'Builder' that keeps track of its size.
+data SizedBuilder = SizedBuilder
```

Comment:
I would wary of the lazy `_size` as well. If we want to maximize encoding speed, I would expect that the approach used by http://hackage.haskell.org/package/cborg with an intermediate custom tailored token stream and a hot encoder loop would perform best. See: https://youtu.be/1WrSPSYhE6o?t=28m23s